src/pulsar_producer.erl

```diff
@@ -283,6 +286,9 @@ idle(info, {Ref, Reply}, State0 = #{lookup_topic_request_ref := Ref}) ->
     State = State0#{lookup_topic_request_ref := undefined},
     erlang:demonitor(Ref, [flush]),
     handle_lookup_topic_reply(Reply, State);
+idle(info, {'EXIT', ParentPid, Reason}, State0 = #{parent_pid := ParentPid}) ->
+    log_error("parent died; shutting down; reason: ~0p", [Reason], State0),
+    {stop, {error, parent_died}};
```

Comment:
Nit. Maybe just?
```suggestion
    {stop, {error, {parent_died, ParentPid, Reason}}};
```
Wondering if we even need that given the parent is supervised most of the time. Otherwise, stop reason is `error`, we should get an error report in the logs later anyway.

Reply:
maybe exit with the same reason.
like linked processes.

Reply:
Now I'm returning `{stop, Reason}`.